packages/shared/src/utils/object-utils.ts

```diff
@@ -77,3 +77,26 @@ export const sortJsonKeys = <T extends Record<string, any>>(json: T): T => {
       return obj;
     }, {}) as T;
 };
+
+/** Minimal deep equality checker, loosely based on lodash _isEqual but for simple primitives only */
+export function isEqual(a: any, b: any) {
+  // handle simple string, boolean, number, undefined, null or same object reference
+  if (a === b) return true;
+  // handle different object types
+  if (typeof a !== typeof b) return false;
+  // handle deep comparison for arrays
+  if (Array.isArray(a)) {
+    const differentEl = a.find((v, i) => !isEqual(v, b[i]));
+    return differentEl ? false : true;
```

Comment:
nit(blocking): This will throw a false positive if a `differentEl` is found that is falsey (e.g. if `a` contains a `0` element at an index that `b` has a different element, e.g. `a = [0]` and `b = [1]`).

How about this instead:
```ts
if (Array.isArray(a)) {
  if (a.length !== b.length) return false;
  return a.every((v, i) => isEqual(v, b[i]));
}
```

Reply:
Really nice catch, you're right `isEqual([true],[false])` would a `true` value. 

I tend to not like `every` operators as there's really no reason to continue processing once you've found an element that differs (can add on a bit of time if dealing with very large arrays). Instead I've added 3a16bc9e3794f6f2baa62a7030feeb3eff0fbe04 which uses the `findIndex` operator to instead focus on the index of the element that differs and ignoring the specific value.